#### Queries on Database

In [1]:
import sqlite3

In [2]:
%load_ext sql

In [3]:
%%sql
sqlite:///CourseDataWareHouse.db

'Connected: @CourseDataWareHouse.db'

#### 1. What are the pre-requisites, co-requisites for course 'Project Management'

In [4]:
%%sql
select CC_ID, PRE_REQS, CO_REQS
from COURSES
where COURSE_TITLE = 'Project Management';

 * sqlite:///CourseDataWareHouse.db
Done.


[(528, 'EC 0278 or MA 0017 or MA 0217.', None),
 (729, None, None),
 (1153, 'IS 0500 or OM 0400.', None)]

EC 0278 or MA 0017 or MA 0217 and IS 0500 or OM 400 are needed as pre reqs for Project Management <br/> Insight: Make sure to take your pre reqs

#### 2. Where were the course meetings for Fundamentals of Business Analytics held in Fall 2018?

In [5]:
%%sql
select distinct MEETINGS.ROOM_NO, COURSE_FACTS.CC_ID, COURSES.COURSE_TITLE 
from COURSE_FACTS, MEETINGS, COURSES
where COURSE_FACTS.CC_ID=COURSES.CC_ID
and COURSE_FACTS.ROOM_ID=MEETINGS.ROOM_ID
and COURSES.COURSE_TITLE = 'Fundamentals of Business Analytics'
and COURSE_FACTS.TERM_NAME = 'Fall2018';

 * sqlite:///CourseDataWareHouse.db
Done.


[('DSB 107', 921, 'Fundamentals of Business Analytics')]

Fundamentals of Business Analytics was held in DSB 107 in the Fall of 2018 <br/> Insight: A lot of classes seem to change rooms/be held in multiple rooms, perhaps a future database can capture mid semester room changes

#### 3. What courses (and number of courses) did Professor Huntley teach in Spring 2019?

In [6]:
%%sql
select distinct COURSE_FACTS.I_ID, COURSES.COURSE_TITLE, PROFESSOR.NAME
from COURSE_FACTS, COURSES, PROFESSOR
where COURSE_FACTS.CC_ID = COURSES.CC_ID
and COURSE_FACTS.I_ID = PROFESSOR.I_ID
and PROFESSOR.NAME like '%Huntley%'
and COURSE_FACTS.TERM_NAME = 'Spring2019';

 * sqlite:///CourseDataWareHouse.db
Done.


[(12869, 'Operations Management', 'Christopher L.  Huntley'),
 (12870, 'Operations Management', 'Christopher L.  Huntley'),
 (12005, 'Databases for Business Analytics', 'Christopher L.  Huntley')]

Professor Huntley taught two Operations Management courses and our Database for Business Analytics class <br/> Insight: Professor Huntley is a busy man!

#### 4. Which classes were held in DSB 105 in Spring 2019?

In [7]:
%%sql
select distinct COURSES.COURSE_TITLE, MEETINGS.ROOM_NO
from COURSE_FACTS, COURSES, MEETINGS
where COURSE_FACTS.CC_ID = COURSES.CC_ID
and COURSE_FACTS.ROOM_ID = MEETINGS.ROOM_ID
and MEETINGS.ROOM_NO = 'DSB 105'
and COURSE_FACTS.TERM_NAME = 'Spring2019';

 * sqlite:///CourseDataWareHouse.db
Done.


[('Entrepreneurship', 'DSB 105'),
 ('Forensic Accounting', 'DSB 105'),
 ('Legal Environment of Business', 'DSB 105'),
 ('Global Competitive Strategy', 'DSB 105'),
 ('Global Capital Markets', 'DSB 105'),
 ('Sports Management', 'DSB 105'),
 ('Intermediate Accounting II', 'DSB 105'),
 ('Introduction to Management Accounting', 'DSB 105'),
 ('Principles of Marketing', 'DSB 105'),
 ('Introduction to Management', 'DSB 105'),
 ('Business Strategies in the Global Environment', 'DSB 105'),
 ('Databases for Business Analytics', 'DSB 105')]

All the courses listed above were taught in DSB 105 this Spring 2019 semester <br/> Insight: 3 accounting classes listed, perhaps accounting professors really like this room and request it

#### 5. Which professors taught the classes for Marketing Research in Spring 2019? 

In [8]:
%%sql
select distinct PROFESSOR.NAME, COURSES.COURSE_TITLE 
from COURSE_FACTS, COURSES, PROFESSOR
where COURSE_FACTS.CC_ID = COURSES.CC_ID
and COURSE_FACTS.I_ID = PROFESSOR.I_ID
and COURSES.COURSE_Title = 'Marketing Research'
and COURSE_FACTS.TERM_NAME = 'Spring2019';

 * sqlite:///CourseDataWareHouse.db
Done.


[('Camelia   Micu', 'Marketing Research'),
 ('Andrew M.  Blum', 'Marketing Research')]

Camelia Micu and Andrew Blum taught Marketing Research in the Spring of 2019 <br/> Insight: Perhaps they specialize in this subject

#### 6. What was the first day of class for the Entrepreneurship course in Spring 2019?

In [10]:
%%sql
select min(TIME.START_TIME), course_title
from COURSE_FACTS, COURSES, TIME
where COURSE_FACTS.TIME_ID = TIME.TIME_ID
and COURSE_FACTS.CC_ID = COURSES.CC_ID
and COURSE_FACTS.TERM_NAME = 'Spring2019'
and COURSE_TITLE = 'Entrepreneurship'
group by course_title;

 * sqlite:///CourseDataWareHouse.db
Done.


[('2019-01-25T08:00:00', 'Entrepreneurship')]

January 25th was their first day of class <br/> Insight: Friday class at 8 AM, students make sure you're on time!

#### 7. Which courses were over capacity in Spring2019? 

In [99]:
%%sql
select distinct COURSES.COURSE_TITLE, (COURSE_FACTS.ACT-COURSE_FACTS.CAP) AS OVER_CAP
from COURSE_FACTS, COURSES
where COURSE_FACTS.CC_ID = COURSES.CC_ID
and COURSE_FACTS.TERM_NAME = 'Spring2019'
GROUP BY COURSE_TITLE
HAVING OVER_CAP > 0;

 * sqlite:///CourseDataWareHouse.db
Done.


[('Abnormal Psychology for Majors', 1),
 ('Accounting Information Systems and Technology', 4),
 ('All About Eve', 3),
 ('Assessment Techniques in Marriage and Family Therapy', 3),
 ('Auditing', 4),
 ('Career Development: Theory and Practice', 3),
 ('Celtic and Early Irish Art', 2),
 ('Conflict Communication', 1),
 ('Counseling Practicum', 1),
 ('Cultural Anthropology', 1),
 ('Customer Behavior', 1),
 ('Databases for Business Analytics', 2),
 ('Design of Mechatronics Systems', 1),
 ('Early America to 1800', 3),
 ('Ethics and Communication', 2),
 ('Explorations in Education: Introduction to Teaching, Learning, and Schooling', 5),
 ('Federal Income Taxation I', 2),
 ('Finite Element Analysis', 1),
 ('Fluid Mechanics', 1),
 ('Gender, Sexuality, and Media', 1),
 ('General Chemistry for Health Science', 1),
 ('General Chemistry for Health Science Lab', 1),
 ('God and Modern Biology', 1),
 ('Graphic Novels as Thriller and Chillers', 1),
 ('Health Communication', 1),
 ('Historical Perspectives on Contemporary Global Crises', 1),
 ('In the Wake of Destruction: Europe Since World War II', 2),
 ('Introduction to Clinical Supervision', 1),
 ('Introduction to Financial Accounting', 1),
 ('Issues in Internal Audit', 2),
 ('Language of Food in Italian Culture', 2),
 ('Modern China: 1800 to Present', 1),
 ('Modern Experimental Methods Lab', 1),
 ('Music of the 20th Century', 4),
 ('News Writing', 1),
 ('Persuasion', 2),
 ('Private Lessons: Piano', 1),
 ('Public Relations', 1),
 ('STEM of Guitar', 1),
 ('Senior Design Project II', 2),
 ('TESOL/Bilingual Student Teaching and DSAP I Seminar', 2),
 ('Texts and Contexts I: Writing as Craft and Inquiry', 1),
 ('The Roman World in Late Antiquity, 284-642 AD', 1),
 ('Violence and the Sacred', 1),
 ('Web Development II with ASP.NET', 3),
 ('World Language Seminar', 1)]

All these courses were over capacity above. <br/> Insight: Maybe Farifield U needs more professors to teach more sections